app/views/login/index.html.haml

```diff
@@ -23,8 +23,8 @@
     %div.d-flex.justify-content-center.flew-wrap
       - omniauth_providers_info.each do |provider, config|
         = button_to  "/auth/#{config[:name] || config[:strategy] || provider}",
-                    class: 'btn btn-lg', style:'width: 70px', title: "Login with #{provider.to_s.humanize}", 'data-turbo': false do
+                    class: 'btn btn-lg',disabled: !config[:enable] ,style:'width: 70px', title: "Login with #{provider.to_s.humanize}", 'data-turbo': false do; 
```

Comment:
if disabled add a tooltip to tell it is disabled

Reply:
To do this I faced some challenges, and here's what I did:
- I used the tooltip component
- The tooltip component already have "icons/icon_name" inside of it so I was forced to remove the "/icons" before the icons in the config file, so the config file became:
`$OMNIAUTH_PROVIDERS = {
  github: {
    client_id: 'CLIENT_ID',
    client_secret: 'CLIENT_SECRET',
    icon: 'github.svg',
    enable: true
  },
  google: {
    strategy: :google_oauth2,
    client_id: 'CLIENT_ID',
    client_secret: 'CLIENT_SECRET',
    icon: 'google.svg',
    enable: true
  },
  orcid: {
    client_id: 'CLIENT_SECRET',
    client_secret: 'CLIENT_SECRET',
    icon: 'orcid.svg',
    enable: false
  },
  keycloak: {
    strategy: :keycloak_openid,
    client_id: 'YOUR_KEYCLOAK_CLIENT_ID',
    client_secret: 'YOUR_KEYCLOAK_CLIENT_SECRET',
    client_options: { site: 'KEYCLOAK_SITE', realm: 'KEYCLOAK_REALM' },
    name: 'keycloak',
    icon: 'keycloak.svg',
    enable: false
  }
}.freeze`
I will add this to the description of the PR so people can access to it later.

Then I needed to add "/icons" in the login view to display them when they are active.

- Another challenge I faced is that in the tooltip we are using an image tag, so we can't change the color of it to show that it is disabled, and when I tried to use inline_svg the tooltip didn't work,
So what I did is that I turned all the svgs colors to grey to look disbaled and then I changed the color of the active onces to look active.

Reply:
OK for now, in the future we will fix the hard-coded `/icon` in the tooltip icon path  argument and  use svg instead of image for the tooltip